CMake fails when an explicit CMAKE_RC_COMPILER is set for the MSVC toolchain

### Description / Steps to reproduce the issue

With the current CMake 3.27.7 release in mingw and msys the following CMake project fails to compile e.g. in the `Developer Command Prompt for Visual Studio 2019` (yes it is not necessary to set CMAKE_RC_COMPILER inside the Developer CMD, it is only for reproducing the error):
CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.25)
project(test)
set(CMAKE_RC_COMPILER "C:/Program Files (x86)/Windows Kits/10/bin/10.0.19041.0/x64/rc.exe")
add_executable(test main.cpp test.rc)
```
main.cpp:
```
int main() { return 0; }
```
test.rc:
```
2 TEXTINCLUDE
BEGIN
    "\0"
END
```
Save files to disk and run:
```
mkdir build
cd build
cmake -GNinja ..
cmake --build .
```
This code works with vanilla CMake 3.27.7, i.e. not from mingw/msys, and it also worked with CMake 3.26.3 from mingw/msys.

### Expected behavior

A successful build.

### Actual behavior

```
C:\Users\XXXXXXXX\git\cmake_rc\build>C:\msys64\mingw64\bin\cmake.exe --build .
[1/3] Building RC object CMakeFiles\test.dir\test.rc.res
FAILED: CMakeFiles/test.dir/test.rc.res
RC C:\Users\XXXXXXXX\git\cmake_rc\test.rc CMakeFiles\test.dir\test.rc.res.d CMakeFiles\test.dir\test.rc.res "Hinweis: Einlesen der Datei: " "C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx86/x86/cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe   -DWIN32 -D_DEBUG /fo CMakeFiles\test.dir\test.rc.res C:\Users\XXXXXXXX\git\cmake_rc\test.rc
Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
Copyright (C) Microsoft Corporation.  All rights reserved.

fatal error RC1107: invalid usage; use RC /? for Help
[2/3] Building CXX object CMakeFiles\test.dir\main.cpp.obj
ninja: build stopped: subcommand failed.
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22621

### MINGW environments affected

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-cmake/PKGBUILD
+++ b/mingw-w64-cmake/PKGBUILD
@@ -8,10 +8,11 @@ pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}"
          $( (( _bootstrap )) || echo \
            "${MINGW_PACKAGE_PREFIX}-ccmake" \
+           "${MINGW_PACKAGE_PREFIX}-${_realname}-cmcldeps" \
            "${MINGW_PACKAGE_PREFIX}-${_realname}-gui" \
            "${MINGW_PACKAGE_PREFIX}-${_realname}-docs"))
 pkgver=3.27.7
-pkgrel=1
+pkgrel=2
 pkgdesc="A cross-platform open-source make system (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -185,6 +186,14 @@ package_cmake() {
     ${MINGW_PREFIX}/bin/emacs -batch -f batch-byte-compile \
       "${pkgdir}${MINGW_PREFIX}"/share/emacs/site-lisp/cmake-mode.el
   fi
+}
+
+package_cmake-cmcldeps() {
+  pkgdesc="CMake wrapper for cl to extract dependencies (mingw-w64)"
+  depends=("${MINGW_PACKAGE_PREFIX}-cmake")
+
+  cd "${srcdir}/build-${MSYSTEM}"
+  DESTDIR="${pkgdir}" ${MINGW_PREFIX}/bin/cmake.exe --install . --component cmcldeps
 }
 
 package_cmake-gui() {

